### DISCUSSION TITLE
What about unset a current context again?

### DISCUSSION THREAD

**Discussion by Schille** [unikubehq/cli#63] [General]
What do you think @buschNT ? Do we need a `project unset`? I was running `project up` awaiting the project selector. But it started the project cluster for a different project (basically the last context).

**Comment by Schille** [comment]
Once you delete a project I would expect the context to be unset as well.

**Reply by Schille** [reply]
@buschNT we where just talking about that.

**Reply by Schille** [reply]
@schwobaseggl what do you think? `unset`? or better yet `context` e.g. `unikube project context --set` or `unikube project context --unset` ?

**Reply by buschNT** [reply]
Maybe we should shift the context into a separate command group? `unikube context`?

Then we could also add something like "show" and "clear all". Optionally we could (at some point) also let users save certain contexts etc.. With `unikube context` it would be all under one group.

Setting/Unsetting context values could be done e.g. like that:
`unikube context set --organization ORGA --project PROJECT --deck DECK`
`unikube context unset --organization ORGA --project PROJECT --deck DECK`

**Reply by Schille** [reply]
@buschNT I think that's a good solution. We may add certain shortcuts in other command groups (such as `deck use`) which can be realized using `context set --deck ...`, too. If a command option is given without a value, e.g. `unikube context set --organization` the interactive selection could be presented as well. However, we may have a problem when setting `unikube context set --project` which then must filter organization projects (if orga context is already set) or allow all projects but with the organization context set implicitly afterwards, right?

**Reply by buschNT** [reply]
> We may add certain shortcuts in other command groups (such as deck use) which can be realized using context set --deck ..., too. 

If required we can do it via an alias. We just have to pay attention that it doesn't get to confusing with too many commands doing the same thing.

> If a command option is given without a value, e.g. unikube context set --organization the interactive selection could be presented as well. 

Yes.

> However, we may have a problem when setting unikube context set --project which then must filter organization projects (if orga context is already set) or allow all projects but with the organization context set implicitly afterwards, right?

This should already be implemented.

**Reply by Schille** [reply]
Let's do that! @buschNT :rocket:
